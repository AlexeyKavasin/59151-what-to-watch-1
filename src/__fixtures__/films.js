export default [
  {
    name: `Fantastic Beasts`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    id: 0,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isPlaying: false,
    genre: `Comedy`
  },
  {
    name: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
    id: 1,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isPlaying: false,
    genre: `Comedy`
  },
  {
    name: `Macbeth`,
    poster: `img/macbeth.jpg`,
    id: 2,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isPlaying: false,
    genre: `Comedy`
  },
  {
    name: `Aviator`,
    poster: `img/Aviator.jpg`,
    id: 3,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isPlaying: false,
    genre: `Comedy`
  },
  {
    name: `Pulp Fiction`,
    poster: `img/pulp-fiction.jpg`,
    id: 4,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isPlaying: false,
    genre: `Documentary`
  }
];
