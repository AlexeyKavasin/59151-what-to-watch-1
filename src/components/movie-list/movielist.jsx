import React from "react";
import PropTypes from "prop-types";
import {MovieCard} from "../movie-card/moviecard.jsx";
import {propTypes as movieCardPropTypes} from "../movie-card/moviecard.props";
import {connect} from "react-redux";
import {ALL_GENRES} from "../../redux/types";

class MovieList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      activeCard: -1
    };
    this.handleMouseOver = this.handleMouseOver.bind(this);
    this.handleMouseLeave = this.handleMouseLeave.bind(this);
  }

  handleMouseOver(event, {id}) {
    const {setActiveItem} = this.props;
    this.setState({
      activeCard: id
    });
    setActiveItem({id});
  }

  handleMouseLeave() {
    const {setActiveItem} = this.props;
    this.setState({
      activeCard: -1
    });
    setActiveItem({id: -1});
  }

  render() {
    const {films} = this.props;
    return <React.Fragment>
      <div className="catalog__movies-list">
        {films.map((film, index) => {
          return <MovieCard
            name={film.name}
            poster={film.poster}
            trailer={film.trailer}
            genre={film.genre}
            key={index}
            id={film.id}
            isPlaying={this.state.activeCard === film.id}
            onMouseOver={this.handleMouseOver}
            onMouseLeave={this.handleMouseLeave}
          />;
        })}
      </div>
    </React.Fragment>;
  }
}

MovieList.propTypes = {
  films: PropTypes.arrayOf(PropTypes.shape(movieCardPropTypes)).isRequired,
  setActiveItem: PropTypes.func,
};

const filterFilms = (films, selectedGenre) => {
  if (selectedGenre === ALL_GENRES) {
    return films;
  }

  return films.filter((film) => film.genre === selectedGenre);
};

const mapStateToProps = (state, ownProps) => ({
  ownProps,
  currentGenre: state.currentGenre,
  genres: state.genres,
  films: filterFilms(state.films, state.currentGenre),
});

const mapDispatchToProps = () => ({});

export {MovieList};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(MovieList);
