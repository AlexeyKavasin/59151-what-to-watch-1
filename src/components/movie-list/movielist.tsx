import * as React from "react";
import {MovieCard} from "../movie-card/moviecard";
import {IMovieList} from "../../interfaces";

export default class MovieList extends React.Component<IMovieList, null> {
  render() {
    const {films, onMouseOver, onMouseLeave, activeCard, filmsToShow} = this.props;
    return <React.Fragment>
      <div className="catalog__movies-list">
        {films.slice(0, filmsToShow).map((film, index) => {
            return <MovieCard
              name={film.name}
              poster={film.preview_image}
              trailer={film.preview_video_link}
              key={index}
              id={film.id}
              isPlaying={activeCard === film.id}
              onMouseOver={onMouseOver}
              onMouseLeave={onMouseLeave}
            />;
        })}
      </div>
    </React.Fragment>;
  }
}
