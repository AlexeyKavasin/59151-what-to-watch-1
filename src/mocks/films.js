export default [
  {
    name: `Fantastic Beasts`,
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
  },
  {
    name: `Bohemian Rhapsody`,
    src: `img/bohemian-rhapsody.jpg`
  },
  {
    name: `Macbeth`,
    src: `img/macbeth.jpg`
  },
  {
    name: `Aviator`,
    src: `img/Aviator.jpg`
  },
  {
    name: `Pulp Fiction`,
    src: `img/pulp-fiction.jpg`
  }
];
