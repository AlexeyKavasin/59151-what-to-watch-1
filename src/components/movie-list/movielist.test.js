import React from 'react';
import renderer from 'react-test-renderer';
import {MovieList} from './movielist.jsx';
import films from '../../__fixtures__/films.js';

it(`Movie list renders correctly`, () => {
  const tree = renderer
    .create(<MovieList
      films={films}
    />, {createNodeMock: () => ({addEventListener() {}, removeEventListener() {}})})
    .toJSON();

  expect(tree).toMatchSnapshot();
});
