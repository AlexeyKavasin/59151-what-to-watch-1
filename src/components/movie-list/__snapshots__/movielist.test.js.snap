// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie list renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    id={0}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      >
        <source
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic Beasts
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id={1}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/bohemian-rhapsody.jpg"
        width="280"
      >
        <source
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id={2}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/macbeth.jpg"
        width="280"
      >
        <source
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Macbeth
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id={3}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/Aviator.jpg"
        width="280"
      >
        <source
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id={4}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/pulp-fiction.jpg"
        width="280"
      >
        <source
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Pulp Fiction
      </a>
    </h3>
  </article>
</div>
`;
